### PR TITLE
Add optional output to result

### PR DESCRIPTION
**Pull request by timja**
cc @garethjevans

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -1,30 +1,45 @@
...
-    exit 1
+    outputResult check-failed 1
   elif [ "$(jq -r .check_run.head_sha < $GITHUB_EVENT_PATH)" \!= $GITHUB_SHA ]
   then
-    echo unexpected commit
-    exit 1
+    outputResult unexpected-commit 1
   else
-    echo passing
+    outputResult passed 0
```
Comment (jglick) [inline run.sh]: Note that this is going to `exit 0` in the original mode, unlike in the previous logic where we fell through to the final `fi`.

File: run.sh [hunk 2]
```diff
@@ -1,30 +1,45 @@
...
+  fi
+}
+
 if [ $GITHUB_EVENT_NAME = check_run ]
 then
   if [ "$(jq -r .check_run.name < $GITHUB_EVENT_PATH)" \!= "$NAME" ]
   then
-    echo wrong check
-    exit 1
+    outputResult wrong-check 1
```
Comment (jglick) [inline run.sh:22]: Does this work with `output_result: true`? The original code would `exit 1` here. Now we are setting `result` to something but falling through, I guess to the final `fi`?

File: run.sh [hunk 3]
```diff
@@ -1,30 +1,45 @@
...
+set -euo pipefail
+
+function outputResult() {
+  local result=$1
+  local exitCode=$2
+
+  echo $result
+
+  if [ $OUTPUT_RESULT = true ]; then
+    echo "::set-output name=result::$result"
```
Comment (jglick) [inline run.sh:11]: Do you not need to include the `exitCode` here? I do not understand what is happening here.

Maybe what you mean is that `result` is either `passed` or some sort of failure message? In that case there should be no `exitCode` arg, and…

File: run.sh [hunk 4]
```diff
@@ -1,30 +1,45 @@
...
+  else
+    exit $exitCode
```
Comment (jglick) [inline run.sh:14]: …this should be
```suggestion
    [ $result = passed ]
    exit $?
```
or something like that.

**Review by jglick — COMMENTED** [review]
Hard to follow. Needs careful testing in both original and new mode in a variety of scenarios. Unfortunately I never figured out how to set up a compelling test environment, so I just did live experiments on a low-risk plugin like `log-cli`.

File: run.sh (hunk 1 above)
Comment (timja) [reply run.sh]: it's an

if
elif
else

it's not going anywhere else?

File: run.sh (hunk 2 above)
Comment (timja) [reply run.sh:22]: added an always exit step

File: run.sh (hunk 3 above)
Comment (timja) [reply run.sh:11]: added, the script should always exit 0 if the exit code is being set as an output

File: run.sh (hunk 4 above)
Comment (timja) [reply run.sh:14]: i think how i've changed it should be fine?

**Comment by timja** [comment]
> Hard to follow. Needs careful testing in both original and new mode in a variety of scenarios. Unfortunately I never figured out how to set up a compelling test environment, so I just did live experiments on a low-risk plugin like `log-cli`.

I tested it fairly extensively by just setting variables in my shell and running locally, adjusting the commit status as required / other vars

File: run.sh (hunk 4 above)
Comment (jglick) [reply run.sh:14]: Works. I think it could be clearer.

**Review by jglick — APPROVED** [review]
So long as it is tested!

**Comment by timja** [comment]
can either of you see what's wrong @garethjevans or @jglick:

https://github.com/jenkinsci/jenkins-infra-test-plugin/runs/2195019274?check_suite_focus=true

the output is evaluating to empty:
https://github.com/jenkinsci/jenkins-infra-test-plugin/blob/master/.github/workflows/cd.yml

**Comment by jglick** [comment]
Disinclination to debug this sort of thing is why I selected a strategy of the minimal number of steps per workflow to begin with. :grin:

**Comment by timja** [comment]
I was missing the ID in the action definition: https://github.com/jenkins-infra/verify-ci-status-action/pull/2/commits/b6ad93ae43b210fcb17c98ffa7abac7c3ded9d69
